### PR TITLE
Add fpslogger

### PR DESCRIPTION
**Pull request by ssysoenko**
Use SPDLOG_FMT_EXTERNAL by default
Move DefaultLoggerFactory to src
Inherit ILogger from enable_shared_from_this
Add ILogger to processArg params
Update spdlog to 1.8.2

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -23,12 +23,13 @@ option(CVSLOGGER_ENABLE_SYSD_BY_DEFAULT "Enable systemd sink by default" ON)
...
-    set(LIB_TYPE SHARED)
-else()
-    set(LIB_TYPE STATIC)
+if (NOT SPDLOG_FMT_EXTERNAL OR NOT SPDLOG_FMT_EXTERNAL_HO)
+    message(NOTICE "Set SPDLOG_FMT_EXTERNAL to ON. "
+        "FPSLogger is not compatible with internal fmt in spdlog version 1.8.2 and older.")
+    set(SPDLOG_FMT_EXTERNAL ON CACHE BOOL "")
 endif()
 
+find_package(fmt REQUIRED)
```
Comment (roman-orekhov) [inline CMakeLists.txt]: won't work if fmt is added to project via add_subdirectory previously

File: CMakeLists.txt [hunk 2]
```diff
@@ -45,15 +46,22 @@ file( GLOB SRCS
         include/cvs/logger/ilogger.h
         include/cvs/logger/loggerfactory.h
         include/cvs/logger/configtypes.h
-        include/cvs/logger/defaultfactory.h
+        include/cvs/logger/tools/fpslogger.hpp
 
-        src/defaultfactory.cpp
+        src/default/defaultfactory.hpp
```
Comment (roman-orekhov) [inline CMakeLists.txt:53]: why mix *.h and *.hpp when both are C++?

File: include/cvs/logger/ilogger.h [hunk 3]
```diff
@@ -27,14 +28,15 @@ class CVSLOGGER_EXPORT ILogger {
   };
 
   template <typename T>
-  typename Strategy<T>::Type processArg(Level, const T& arg) {
+  typename Strategy<T>::Type processArg(const std::shared_ptr<ILogger>&, Level, const T& arg) {
```
Comment (roman-orekhov) [inline include/cvs/logger/ilogger.h]: I don't see a use case for this, not in this commit for sure. Don't think it's a necessary change for this commit -- better to include it with the change where it's needed or in a separate commit right before such change

**Review by roman-orekhov — CHANGES_REQUESTED** [review]
Generally I'd prefer splitting commits into parts dealing with particular additions: some build-only fixes, some about feature1, some about feature2, etc

File: include/cvs/logger/ilogger.h (hunk 3 above)
Comment (ssysoenko) [reply include/cvs/logger/ilogger.h]: Difference? In the release, messages will still get mixed up.

Comment (roman-orekhov) [reply include/cvs/logger/ilogger.h]: > 
> 
> Difference? In the release, messages will still get mixed up.

I don't get the answer. What can be different if the parameter isn't used anywhere? Is it just to keep shared_ptr alive at certain use case? Can you describe the use case?

**Review by roman-orekhov — APPROVED** [review]

**Review by Aldrog — APPROVED** [review]

**Pull request merged**